draft-cms-masque-ip-proxy-reqs.md

```diff
@@ -152,6 +152,8 @@ along the path.
 
 The server will assign an IP address of its choosing to the client. The client
 may suggest a preference of IP address which the server can choose to follow.
+Similarly, to support the network-to-network use case, the client will be able
```

Comment:
This is not how I would expect the N2N case to work. Both sides have preexisting IPs and should just be advertising routes to each other.

Reply:
From your email, I thought that for N2N we need the client to able to assign the server an IP from the client network. If not, why do we need this?

Reply:
There's actually a few cases here, some of which are n2n and some of which are p2n.

For p2n, there's two operating modes we want:
* Give me a (new) IP address
* I prefer a specific IP address, if available

The latter is useful if you had a more dynamic assignment model and the VPN client re-connected, in order to keep the same IP assignment and avoid resetting TCP/QUIC connections over the tunnel.

For n2n, the operating mode will be one of
* Delegate a (new) IP range to the client network (but I anticipate this to be rare)
* Client announces an existing IP range

Reply:
Discussed offline, let's keep the IP Assignment and Route Negotiation requirements separate, even if we won't need to use the entire cross-product of possibilities.